### PR TITLE
[lte][agw] Adding role in ansible to install and build ghz load test

### PR DESCRIPTION
**Pull request by ardzoht**
Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

## Summary

- Adds ghz ansible steps to install tool to launch load tests for gRPC services 

## Test Plan

- tested locally to ensure ghz is properly installed and called 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/deploy/roles/ghz/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ghz_root: /var/tmp/ghz
```
Comment (koolzz) [inline lte/gateway/deploy/roles/ghz/defaults/main.yml]: not sure if /var/tmp is the best place for this

**Review by koolzz — APPROVED** [review]

File: lte/gateway/deploy/roles/ghz/files/build_ghz.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# shellcheck disable=SC1091
+source /etc/environment
```
Comment (tmdzk) [inline lte/gateway/deploy/roles/ghz/files/build_ghz.sh]: As building is 3 commands, let's move it to ansible.

File: lte/gateway/deploy/roles/ghz/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+###############################################
+# Download and build gHZ (load test dependency)
+###############################################
+- name: Clone ghz repo
+  become: yes
+  git:
+    repo: 'https://github.com/bojand/ghz'
+    dest: "{{ ghz_root }}"
+    version: v0.94.0
```
Comment (tmdzk) [inline lte/gateway/deploy/roles/ghz/tasks/main.yml]: Turn that into a variable

File: lte/gateway/deploy/roles/ghz/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################
+# Download and build gHZ (load test dependency)
+###############################################
+- name: Clone ghz repo
+  become: yes
```
Comment (tmdzk) [inline lte/gateway/deploy/roles/ghz/tasks/main.yml]: If you change the ghz root folder variable to /tmp you should be able to execute it without being `root` . It's just that `/var` is owned by root

File: lte/gateway/deploy/roles/ghz/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  git:
+    repo: 'https://github.com/bojand/ghz'
+    dest: "{{ ghz_root }}"
+    version: v0.94.0
+    force: yes
+  when: preburn
```
Comment (tmdzk) [inline lte/gateway/deploy/roles/ghz/tasks/main.yml]: If I remember correctly, that means we have to build a new Vagrant image, provisioning wont' be enough

File: lte/gateway/deploy/roles/ghz/defaults/main.yml (hunk 1 above)
Comment (tmdzk) [reply lte/gateway/deploy/roles/ghz/defaults/main.yml]: +1 

Use `/tmp/ghz`

**Review by tmdzk — CHANGES_REQUESTED** [review]
Idea: can we just build it once, upload it to the registry and just pull the binary from there?

File: lte/gateway/deploy/roles/dev_common/tasks/main.yml [hunk 6]
```diff
@@ -354,6 +354,21 @@
     CODEGEN_ROOT: "{{ codegen_root }}"
   when: preburn
 
+- name: Download gHZ binary from artifactory
+  get_url:
+    url: https://artifactory.magmacore.org:443/artifactory/blob-prod/ghz
+    dest: "/tmp"
```
Comment (tmdzk) [inline lte/gateway/deploy/roles/dev_common/tasks/main.yml]: you can directly get the binary at the right sport using `dest: /usr/local/bin/ghz` and mode to set up the correct rights

**Review by tmdzk — APPROVED** [review]

**Review by pshelar — APPROVED** [review]
Thanks!

**Pull request merged**